### PR TITLE
[TFLite] Add int8 and int16x8 support for the EXP operator

### PR DESCRIPTION
**Pull request by Tessil**
Hello,

This PR adds quantized int8 and int16 support for the EXP operator. It uses a lookup table to do so.

Thanks,
Thibaut

### CODE REVIEW

**Review by abattery — CHANGES_REQUESTED** [review]
Could you add e2e test cases under https://github.com/tensorflow/tensorflow/tree/master/tensorflow/lite/testing/op_tests ?

**Comment by Tessil** [comment]
Thanks for the review.

Aren't the changes in https://github.com/tensorflow/tensorflow/pull/58278/files#diff-49276b87644be533e959f64f5f3ad6dcd2bab42a2391c2b2d27c20d08ac6cdc5R29 sufficient for op_tests? It is similar to https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/testing/op_tests/relu.py tests.

**Comment by abattery** [comment]
@alankelly in the terms of quantized kernel wise, it looks okay to me.

Could you review this change as the TFLite kernel view point?

**Comment by gbaned** [comment]
Hi @alankelly Can you please review this PR ? Thank you!

**Review by alankelly — APPROVED** [review]

File: tensorflow/lite/kernels/exp.cc [hunk 1]
```diff
@@ -37,30 +39,77 @@ struct ExpContext {
...
+    TF_LITE_ENSURE_EQ(context, output->params.zero_point, 0);
+  }
+
+  if (input->type == kTfLiteInt8) {
+    LUTPopulate<int8_t>(input->params.scale, input->params.zero_point,
+                        output->params.scale, output->params.zero_point,
+                        [](float value) { return std::exp(value); },
+                        data->lut_int8);
+  } else if (input->type == kTfLiteInt16) {
+    LUTPopulate<int16_t>(input->params.scale, input->params.zero_point,
```
[the remark this answers is not in the document]
Comment (Tessil) [reply tensorflow/lite/kernels/exp.cc:78]: I merged the above zero-point check in the else if instead, hope it's alright so. Thanks.

**Comment by gbaned** [comment]
Hi @alankelly Can you please review this PR ? Thank you!

**Comment by alankelly** [comment]
Can you please fix the conflict and I will merge tomorrow?

**Comment by Tessil** [comment]
@alankelly Thank you, I fixed the conflict.

**Review by alankelly — APPROVED** [review]

**Comment by alankelly** [comment]
Thanks, started the merge. Reduce should also land today once the Americans wake up and approve.

**Pull request merged**